packages/apps/kadena-docs/src/components/Layout/Full/Full.tsx

```diff
@@ -7,48 +7,96 @@ import {
   AsideLink,
   AsideList,
   Content,
+  ListItem,
   StickyAside,
   StickyAsideWrapper,
 } from '../components';
 
 import { BottomPageSection } from '@/components/BottomPageSection';
 import { ILayout, ISubHeaderElement } from '@/types/Layout';
 import { createSlug } from '@/utils';
-import React, { FC, ReactNode } from 'react';
+import { analyticsEvent, EVENT_NAMES } from '@/utils/analytics';
+import { useRouter } from 'next/router';
+import React, {
+  FC,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 
-const renderListItem = (item: ISubHeaderElement): ReactNode => {
-  if (item.title === undefined) return null;
+export const Full: FC<ILayout> = ({ children, aSideMenuTree = [] }) => {
+  const scrollRef = useRef<HTMLDivElement | null>(null);
+  const router = useRouter();
+  const menuRef = useRef<HTMLUListElement | null>(null);
+  const [activeItem, setActiveItem] = useState<string>('');
 
-  const slug = createSlug(item.title);
-  return (
-    <AsideLink href={`#${slug}`} key={slug} label={item.title}>
-      {item.children.length > 0 && (
-        <AsideList inner={true}>{item.children.map(renderListItem)}</AsideList>
-      )}
-    </AsideLink>
-  );
-};
+  const updateEntry = ([entry]: IntersectionObserverEntry[]): void => {
+    const { isIntersecting } = entry;
+
+    if (isIntersecting) {
+      setActiveItem(entry.target.getAttribute('href') ?? '');
+    }
+  };
+
+  useEffect(() => {
+    if (activeItem === null) {
+      const hashPath =
+        router.asPath.split('#').length === 2
```

Comment:
Maybe in 2 steps?

const [,hash] = router.asPath.split('#')
const hashPath = hash ? '#'+hash : ''

Reply:
that is better, yes.